crates/swc_ecma_transforms_compat/src/es2022/class_properties/private_field.rs

```diff
@@ -40,21 +42,29 @@ impl PrivateRecord {
         self.0.pop();
     }
 
-    pub fn get(&self, name: &JsWord) -> (Mark, PrivateKind, &Ident) {
+    pub fn get(&self, name: &Ident) -> (Mark, PrivateKind, &Ident) {
         for p in self.0.iter().rev() {
-            if let Some(kind) = p.ident.get(name) {
-                return (p.mark, kind.clone(), &p.class_name);
+            if let Some(kind) = p.ident.get(&name.sym) {
+                return (p.mark, *kind, &p.class_name);
             }
         }
-        // TODO: better error information with span
-        panic!("Private name #{} is not defined.", name);
+
+        let error = format!("private name #{} is not defined.", name.sym);
+        HANDLER.with(|handler| handler.struct_span_err(name.span, &error).emit());
+        (
+            Mark::from_u32(1),
```

Comment:
Why did you use this? I think `Mark::root` would be better

Reply:
`Mark::root` would panic at https://github.com/swc-project/swc/blob/main/crates/swc_common/src/syntax_pos/hygiene.rs#L345

Reply:
If so, I think the code calling `SyntaxContext.apply` should check if it's a root mark and avoid panic.
(I mean the caller of this `get` method)

Reply:
Yes, thank you so much!